tracer/src/Datadog.Trace.SourceGenerators/Configuration/ConfigKeyAliasesSwitcherGenerator.cs

```diff
@@ -0,0 +1,386 @@
+// <copyright file="ConfigKeyAliasesSwitcherGenerator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using Datadog.Trace.SourceGenerators;
+using Datadog.Trace.SourceGenerators.Helpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+/// <summary>
+/// Source generator that reads supported-configurations.json and generates a switch case
+/// for configuration key matching with alias support.
+/// </summary>
+[Generator]
+public class ConfigKeyAliasesSwitcherGenerator : IIncrementalGenerator
+{
+    private const string SupportedConfigurationsFileName = "supported-configurations.json";
+    private const string MainKeyParamName = "mainKey";
+    private const string ClassName = "ConfigKeyAliasesSwitcher";
+
+    /// <inheritdoc />
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        // Get the supported-configurations.json file and parse only the aliases section
+        // We only track changes to the aliases section since that's what affects the generated code
+        var additionalText = context.AdditionalTextsProvider
+                                    .Where(static file => Path.GetFileName(file.Path).Equals(SupportedConfigurationsFileName, StringComparison.OrdinalIgnoreCase))
+                                    .WithTrackingName(TrackingNames.ConfigurationKeysAdditionalText);
+
+        var aliasSection = additionalText.Collect()
+                                        .Select(static (files, ct) =>
+                                        {
+                                            if (files.Length == 0)
```

Comment:
I'm not sure this will ever fire 🤔 Given we filter out non-matching texts, I don't think it will. So we can probably just remove the `Collect`, remove the diagnostic, and make this `Select` operate on a single file